pkgs/development/compilers/cppfront/default.nix

```diff
@@ -0,0 +1,43 @@
+{ stdenv
+, fetchFromGitHub
+, lib
+}:
+
+stdenv.mkDerivation {
+  pname = "cppfront";
+  version = "unstable-2023-05-15";
+
+  src = fetchFromGitHub {
+    repo = "cppfront";
+    owner = "hsutter";
+    rev = "ebe5ebc11303b11d29d6d40d498372c2bea07abd";
+    sha256 = "sha256-PtRmN2HsDC6byXWR2iFepvEg1KBNeIsYm/LKqlFlZ7Y=";
+  };
+
+  buildPhase = ''
+    runHook preBuild
+    $CXX source/cppfront.cpp -std=c++20 -o cppfront
```

Comment:
there should be a build in method for setting flags, if you are just overriding the buildPhase to set flags.
for cmake it would be `cmakeFlags = [ ]` not quite sure what you would need in this case.

Reply:
This package doesn't have a build system, this is the official build instruction.